src/useSlider.ts

```diff
@@ -0,0 +1,150 @@
+      const reverse = options.reverse === undefined ? false : options.reverse;
+
+      if (ref.current && styles) {
+        ref.current.style.userSelect = 'none';
+      }
+
+      const startScrubbing = () => {
+        if (!isSliding.current && isMounted()) {
+          (options.onScrubStart || noop)();
+          isSliding.current = true;
+          setState({ isSliding: true });
+          bindEvents();
+        }
+      };
+
+      const stopScrubbing = () => {
+        if (isSliding.current && isMounted()) {
+          (options.onScrubStop || noop)();
+          isSliding.current = false;
+          setState({ isSliding: false });
+          unbindEvents();
+        }
+      };
+
+      const onMouseDown = (event: MouseEvent) => {
+        startScrubbing();
+        onMouseMove(event);
+      };
+      const onMouseMove = (event: MouseEvent) => onScrub(event.clientX);
+
+      const onTouchStart = (event: TouchEvent) => {
+        startScrubbing();
+        onTouchMove(event);
+      };
+      const onTouchMove = (event: TouchEvent) => onScrub(event.changedTouches[0].clientX);
+
+      const bindEvents = () => {
+        on(document, 'mousemove', onMouseMove);
+        on(document, 'mouseup', stopScrubbing);
+
+        on(document, 'touchmove', onTouchMove);
+        on(document, 'touchend', stopScrubbing);
+      };
+
+      const unbindEvents = () => {
+        off(document, 'mousemove', onMouseMove);
+        off(document, 'mouseup', stopScrubbing);
+
+        off(document, 'touchmove', onTouchMove);
+        off(document, 'touchend', stopScrubbing);
+      };
+
+      const onScrub = (clientX: number) => {
+        cancelAnimationFrame(frame.current);
+
+        frame.current = requestAnimationFrame(() => {
+          if (isMounted() && ref.current) {
+            const { left: pos, width: length } = ref.current.getBoundingClientRect();
+
+            // Prevent returning 0 when element is hidden by CSS
+            if (!length) {
+              return;
+            }
+
+            let value = (clientX - pos) / length;
+
+            if (value > 1) {
+              value = 1;
+            } else if (value < 0) {
+              value = 0;
+            }
+
+            if (reverse) {
+              value = 1 - value;
+            }
+
+            setState({
+              value,
+              pos: clientX - pos,
+              length,
+            });
+
+            (options.onScrub || noop)(value);
+          }
+        });
+      };
+
+      on(ref.current, 'mousedown', onMouseDown);
+      on(ref.current, 'touchstart', onTouchStart);
+
+      return () => {
+        off(ref.current, 'mousedown', onMouseDown);
+        off(ref.current, 'touchstart', onTouchStart);
+      };
+    } else {
+      return undefined;
+    }
+  }, [ref]);
+
+  useEffect(() => {
+    setState(prevState => ({
+      pos: Math.round(prevState.value * width),
+      length: width,
+    }));
+  }, [width]);
```

Comment:
Maybe we should leave tracking the re-sizing up to the user. Users receives the `value`, which is normalized to `[0 ... 1]` and they can figure out what to do with it. Also, they can use `%` units in CSS with `value`, see proposed change to the story. Then we don't need to use `useMeasureDirty()`, which uses `ResizeObserver`, which is not supported by some older browsers.